File: src/lessons/welcome/array/island/IslandEntity.py
from RemoteBat import *
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(island(param[0])))

# BEGIN TEMPLATE
def island(num):
# BEGIN SOLUTION
  nbisland=0
  for i in range(len(num)-1):
    if num[i]<num[i+1]:
      nbisland=nbisland+1
  return nbisland
# END SOLUTION
# END TEMPLATE

File: src/lessons/welcome/array/island/IslandEntity.java
package lessons.welcome.array.island;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class IslandEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(island((int[]) param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  int island(int[] num)
  {
    /* BEGIN SOLUTION */
    int nbisland = 0;
    for (int i = 0; i < num.length - 1; i++) {
      if (num[i] < num[i + 1]) {
        nbisland++;
      }
    }
    return nbisland;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}
